src/Runtime.Base/src/System/Runtime/RuntimeExports.cs

```diff
@@ -186,13 +181,10 @@ public static unsafe void RhUnboxAny(object o, ref byte data, EETypePtr pUnboxTo
         public static unsafe ref byte RhUnbox2(EETypePtr pUnboxToEEType, Object obj)
         {
             EEType* ptrUnboxToEEType = (EEType*)pUnboxToEEType.ToPointer();
-            if (obj.EEType != ptrUnboxToEEType)
+            if ((obj == null) || ((obj.EEType != ptrUnboxToEEType) && !UnboxAnyTypeCompare(obj.EEType, ptrUnboxToEEType)))
```

Comment:
Do we get anything out of the `(obj.EEType != ptrUnboxToEEType)` part? RyuJIT seems to be inlining this check and will skip `RhUnbox2` call completely if this is true. It will also conveniently trigger a hardware AV if `obj` is null.